core/chain_manager.go

```diff
@@ -335,23 +319,23 @@ func (self *ChainManager) ExportN(w io.Writer, first uint64, last uint64) error
 // insert injects a block into the current chain block chain. Note, this function
 // assumes that the `mu` mutex is held!
 func (bc *ChainManager) insert(block *types.Block) {
-	err := WriteHead(bc.chainDb, block)
-	if err != nil {
-		glog.Fatal("db write fail:", err)
+	// Insert the block as the new head of the chain
```

Comment:
```WriteCanonicalHash``` seems to store a hash under a block number. Does it store any information about the new head of the chain?